src/carma-platform/guidance_scripts.py

```diff
@@ -1283,6 +1283,352 @@ def get_planner_trajectory_intervals(
+        mcap_path,
+        topics,
+        start_time=start_time,
+        end_time=end_time,
+        field_extractors={
+            topics[0]: lambda msg: (
+                msg.twist.linear.x  # longitudinal velocity
+            ),
+            topics[1]: lambda msg: (
+                msg.speed_limit     # Actual speed limit in m/s
+            ),
+            topics[2]: lambda msg: (
+                msg.cmd.linear_velocity # Commanded speed in m/s
+            )
+        },
+    )
+
+    # Extract time series data
+    timestamps_vel, velocity_data = extracted_data[topics[0]]
+    long_velocities = velocity_data
+
+    timestamps_speed_limit, speed_limit_data = extracted_data[topics[1]]
+    speed_limits =  speed_limit_data
+
+    timestamps_cmd, cmd_data = extracted_data[topics[2]]
+    cmd_velocities = cmd_data
+
+    # Approximate aligned timestamps for all variables for plotting
+    # Align time series data for consistent comparison using existing function twice
+    first_aligned_timestamps, first_aligned_velocities, aligned_speed_limits = align_time_series(
+        timestamps_vel, long_velocities,
+        timestamps_speed_limit, speed_limits
+    )
+
+    # Second alignment to get all three series aligned
+    timestamps, aligned_cmd_velocities, aligned_velocities = align_time_series(
+        timestamps_cmd, cmd_velocities,
+        first_aligned_timestamps, first_aligned_velocities
+    )
+
+    # Now get aligned speed limits with the same timestamps
+    aligned_speed_limits = np.interp(timestamps, first_aligned_timestamps, aligned_speed_limits)
+
+    # Reassign to variables used later in the function
+    long_velocities = aligned_velocities
+    speed_limits = aligned_speed_limits
+    cmd_velocities = aligned_cmd_velocities
+
+    # Detect speed limit changes only use raw speed limit data for accuracy
+    speed_limit_changes = detect_speed_limit_changes(timestamps_speed_limit, speed_limit_data)
+
+    # Analyze response to speed changes only use raw cmd data for accuracy
+    response_times, steady_state_periods = analyze_speed_responses(
+        timestamps_cmd,
+        cmd_data,
+        speed_limit_changes,
+        steady_state_duration,
+        speed_tolerance_pct
+    )
+
+    # Calculate statistics
+    statistics = {
+        "num_speed_changes": len(speed_limit_changes),
+        "response_times": {
+            "mean": float(np.mean(response_times)) if len(response_times) > 0 else None,
+            "median": float(np.median(response_times)) if len(response_times) > 0 else None,
+            "min": float(np.min(response_times)) if len(response_times) > 0 else None,
+            "max": float(np.max(response_times)) if len(response_times) > 0 else None,
+            "values": response_times.tolist() if len(response_times) > 0 else []
+        },
+        "steady_state": {
+            "total_periods": len(steady_state_periods),
+            "periods": steady_state_periods
```

Comment:
Technically these are not the steady state periods, they are just the first X seconds of steady state after a speed limit change (based on `steady_state_duration`). Not exactly sure how we reword this, but I will think on it

Reply:
I renamed to steady_state_confirmation_period